src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurRequestExtractFileServiceImpl.java

```diff
@@ -0,0 +1,112 @@
+    protected ConcurRequestExtractFileValidationService concurRequestExtractFileValidationService;
+
+    public List<String> getUnprocessedRequestExtractFiles() {
+        List<String> fileNamesToLoad = getBatchInputFileService().listInputFileNamesWithDoneFile(getBatchInputFileType());
+        return fileNamesToLoad;
+    }
+
+    public boolean processFile(String requestExtractFileName) {
+        boolean processingSuccess = true;
+        try {
+            Object parsedFile = loadFile(requestExtractFileName);
+            processingSuccess = requestExtractHeaderRowValidatesToFileContents(parsedFile);
+        } catch (RuntimeException re) {
+            LOG.error("Caught exception trying to load: " + requestExtractFileName, re);
+            processingSuccess = false;
+        }
+        return processingSuccess;
+    }
+
+    public boolean requestExtractHeaderRowValidatesToFileContents(Object parsedFile) {
+        boolean headerValidationPassed = false;
+        List<ConcurRequestExtractFile> requestExtractFiles = (ArrayList<ConcurRequestExtractFile>) parsedFile;
+        if (requestExtractFiles.isEmpty() || requestExtractFiles.size() != 1) {
+            LOG.error("Header validation should have received single parsed file. More or less than one parsed file was detected.");
+        }
+        else {
+            ConcurRequestExtractFile requestExtractFile = requestExtractFiles.get(0);
+            LOG.info("batchDate=" + requestExtractFile.getBatchDate() + "=   recordCount=" + requestExtractFile.getRecordCount() + "=     totalApprovedAmount=" + requestExtractFile.getTotalApprovedAmount() + "=");
+            headerValidationPassed = (getConcurRequestExtractFileValidationService().fileRowCountMatchesHeaderRowCount(requestExtractFile) &&
+                                      getConcurRequestExtractFileValidationService().fileApprovedAmountsMatchHeaderApprovedAmount(requestExtractFile));
+        }
+        return headerValidationPassed;
+    }
+
+    public void setBatchInputFileService(BatchInputFileService batchInputFileService) {
+        this.batchInputFileService = batchInputFileService;
+    }
+
+    public BatchInputFileService getBatchInputFileService() {
+        return batchInputFileService;
+    }
+
+    public void setBatchInputFileType(BatchInputFileType batchInputFileType) {
+        this.batchInputFileType = batchInputFileType;
+    }
+
+    public BatchInputFileType getBatchInputFileType() {
+        return batchInputFileType;
+    }
+
+    public void setConcurRequestExtractFileValidationService(ConcurRequestExtractFileValidationService concurRequestExtractFileValidationService) {
+        this.concurRequestExtractFileValidationService = concurRequestExtractFileValidationService;
+    }
+
+    public ConcurRequestExtractFileValidationService getConcurRequestExtractFileValidationService() {
+        return this.concurRequestExtractFileValidationService;
+    }
+
+    private Object loadFile(String fileName) {
```

Comment:
I believe that this should be left as an object since method loadFile will be placed in a utility class with user story KFSPTS-7997 and will need to be used by multiple file types.